ui/src/components/SettingsModal/SettingModal.tsx

```diff
@@ -0,0 +1,146 @@
+import spinner from "../assets/spinner.svg";
+import {useCommonStyles} from "../../helpers/commonStyle";
+import {toast} from "react-toastify";
+
+interface SettingsModalProps {
+    isOpen: boolean
+    onClose: () => void
+    isFirstLogin: boolean
+}
+
+const api = Api.getInstance();
+
+export const SettingsModal: React.FC<SettingsModalProps> = ({isOpen, onClose, isFirstLogin}) => {
+
+    const classes = useCommonStyles();
+    const [namespaces, setNamespaces] = useState({});
+    const [isLoading, setIsLoading] = useState(false);
+    const [searchValue, setSearchValue] = useState("");
+
+    useEffect(() => {
+        (async () => {
+            try {
+                setIsLoading(true);
+                const tapConfig = await api.getTapConfig()
+                if(isFirstLogin) {
+                    const namespacesObj = {...tapConfig?.tappedNamespaces}
+                    Object.keys(tapConfig?.tappedNamespaces ?? {}).forEach(namespace => {
+                        namespacesObj[namespace] = true;
+                    })
+                    setNamespaces(namespacesObj);
+                } else {
+                    setNamespaces(tapConfig?.tappedNamespaces);
+                }
+            } catch (e) {
+                console.error(e);
+            } finally {
+                setIsLoading(false);
+            }
+        })()
+    }, [isFirstLogin])
+
+    const setAllNamespacesTappedValue = (isTap: boolean) => {
+        const newNamespaces = {};
+        Object.keys(namespaces).forEach(key => {
+            newNamespaces[key] = isTap;
+        })
+        setNamespaces(newNamespaces);
+    }
+
+    const updateTappingSettings = async () => {
+        try {
+            await api.setTapConfig(namespaces);
+            onClose();
+            toast.success("Saved successfully");
+        } catch (e) {
+            console.error(e);
+            toast.error("Something went wrong, changes may not have been saved.")
+        }
+    }
+
+    const toggleTapNamespace = (namespace) => {
+        const newNamespaces = {...namespaces};
+        newNamespaces[namespace] = !namespaces[namespace]
+        setNamespaces(newNamespaces);
+    }
+
+    const toggleAll = () => {
+        const isChecked = Object.values(namespaces).every(tap => tap === true);
+        isChecked ? setAllNamespacesTappedValue(false) : setAllNamespacesTappedValue(true);
+    }
+
+    const buildNamespacesTable = () => {
+        return <table cellPadding={5} style={{borderCollapse: "collapse"}}>
+            <thead>
+            <tr style={{borderBottomWidth: "2px"}}>
+                <th style={{width: 50}}><Checkbox checked={Object.values(namespaces).every(tap => tap === true)}
+                                                  onToggle={toggleAll}/></th>
+                <th>Namespace</th>
+            </tr>
+            </thead>
+            <tbody>
+            {Object.keys(namespaces).filter((namespace) => namespace.includes(searchValue)).map(namespace => {
```

Comment:
Might be worth it to do something like `const filteredNamespaces = useMemo(...` for performance wdyt?